tests/integration/cli/setup_test.py

```diff
@@ -0,0 +1,51 @@
+from mock import patch
+
+from ...testcases import DustyIntegrationTestCase
+
+from dusty import constants
+from dusty.config import save_config_value
+
+class TestSetupCLI(DustyIntegrationTestCase):
+    def setUp(self):
+        super(TestSetupCLI, self).setUp()
+        save_config_value(constants.CONFIG_SETUP_KEY, False)
+        save_config_value(constants.CONFIG_SPECS_REPO_KEY, '')
+        save_config_value(constants.CONFIG_MAC_USERNAME_KEY, 'nobody')
+
+    def tearDown(self):
+        super(TestSetupCLI, self).tearDown()
+
+    @patch('dusty.commands.setup._get_recommended_vm_size')
+    @patch('dusty.commands.setup._get_raw_input')
+    def test_setup_defaults(self, fake_raw_input, fake_vm_size):
+        fake_vm_size.return_value = 2048
+        fake_raw_input.side_effect = ['y', '', 'y']
+        self.run_command('setup')
+        self.assertConfigValue(constants.CONFIG_MAC_USERNAME_KEY, self.current_user)
+        self.assertConfigValue(constants.CONFIG_SPECS_REPO_KEY, 'github.com/gamechanger/dusty-example-specs')
+        self.assertConfigValue(constants.CONFIG_VM_MEM_SIZE, 2048)
+
+    @patch('dusty.commands.setup._get_raw_input')
+    def test_setup_override_user(self, fake_raw_input):
+        fake_raw_input.side_effect = ['n', self.current_user, '', 'y']
+        self.run_command('setup')
+        self.assertConfigValue(constants.CONFIG_MAC_USERNAME_KEY, self.current_user)
+
+    @patch('dusty.commands.setup._get_raw_input')
+    def test_setup_override_specs(self, fake_raw_input):
+        fake_raw_input.side_effect = ['y', 'github.com/gamechanger/dusty-example-specs', 'y']
```

Comment:
i think it is a more thorough test to check that both this one and the test above set the values to something other then self.current_user and dusty-example-specs.  That means that across the first and the second two tests we are ensuring that we are not just always writing those values to the config in setup.